engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/impl/TaskRestServiceImpl.java

```diff
@@ -96,8 +100,19 @@ public List<TaskDto> queryTasks(TaskQueryDto queryDto, Integer firstResult,
     TaskQuery query = queryDto.toQuery(engine);
 
     List<Task> matchingTasks = executeTaskQuery(firstResult, maxResults, query);
-
     List<TaskDto> tasks = new ArrayList<TaskDto>();
+
```

Comment:
Line 104 - 115 seems unececcerly complicated. I think this would mean the same: 
```
    boolean withTaskVars = Boolean.TRUE.equals(queryDto.getWithTaskVariablesInReturn());
    boolean withTaskLocalVars = Boolean.TRUE.equals(queryDto.getWithTaskLocalVariablesInReturn());
    boolean withCommentInfo = Boolean.TRUE.equals(queryDto.getWithCommentAttachmentInfo());

    if (withTaskVars || withTaskLocalVars){
        return getVariablesForTasks(engine, matchingTasks, withTaskVars, withCommentInfo);
    }
```